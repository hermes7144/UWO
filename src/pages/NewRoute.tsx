import React, { useEffect } from 'react';
import Button from '../components/ui/Button';
import { useNavigate } from 'react-router-dom';
import useRoute from '../Hooks/useRoute';
import { useAuthContext } from '../context/AuthContext';
import Map from '../components/Map';
import Citys from '../components/Citys';
import RouteForm from '../components/RouteForm';
import { useUWORouteContext } from '../context/UWORouteContext';

export default function NewRoute() {
  const { setEditable, setCitys } = useUWORouteContext();
  const { uid } = useAuthContext();
  const { removeItem } = useRoute();
  const navigate = useNavigate();

  const {
    routeQuery: { data: route },
  } = useRoute();

  useEffect(() => {
    route.citys ? setCitys(route.citys) : setCitys([]);
    setEditable(true);

    return () => {
      setEditable(false);
    };
  }, [route, setCitys, setEditable]);

  const handleDeleteRoute = () => {
    if (!window.confirm('삭제하시겠습니까?')) return false;
    const id = route.id;
    removeItem.mutate({ id }, { onSuccess: () => navigate('/routes') });
  };

  return (
    <div className='flex flex-col sm:flex-row'>
      <div className='basis-4/6'>
        <Map />
      </div>
      <div className='basis-2/6 flex flex-col p-2'>
        <div className='flex justify-end'>{route && route.user_id === uid && <Button text={'삭제'} onClick={handleDeleteRoute} />}</div>
        <Citys route={route} />
        <RouteForm />
      </div>
    </div>
  );
}
