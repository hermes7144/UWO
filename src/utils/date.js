// import { format, register } from 'timeago.js';
// import koLocale from 'timeago.js/lib/lang/ko';

// register('ko', koLocale);

// export function formatAgo(date, lang = 'en_US') {
//   return format(date, lang);
// }

export function formatDate(rawDate) {
  const date = new Date(rawDate);
  const year = date.getFullYear().toString().slice(-2); // 연도에서 뒤의 두 자리만 추출
  const month = ('0' + (date.getMonth() + 1)).slice(-2); // 월을 두 자리 숫자로 변환
  const day = ('0' + date.getDate()).slice(-2); // 일을 두 자리 숫자로 변환
  return `${year}/${month}/${day}`;
}