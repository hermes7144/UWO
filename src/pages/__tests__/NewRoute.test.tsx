import { render } from '@testing-library/react';
import { withAllContexts, withRouter } from '../../tests/utils';
import { Route } from 'react-router-dom';
import NewRoute from '../NewRoute';
import { fakeRoute } from '../../tests/fakeRoutes';

describe('NewRoute', () => {
  const fakeUseRoute = jest.fn();

  beforeEach(() => {
    fakeUseRoute.mockImplementation(() => {
      return {
        routeQuery: { isLoading: false, data: fakeRoute },
      };
    });
  });

  afterEach(() => {
    fakeUseRoute.mockReset();
  });

  it('renders correctly', () => {
    const { asFragment } = renderNewRoute();
    expect(asFragment()).toMatchSnapshot();
  });

  function renderNewRoute() {
    return render(withAllContexts(withRouter(<Route path='/' element={<NewRoute />} />), fakeUseRoute));
  }
});
