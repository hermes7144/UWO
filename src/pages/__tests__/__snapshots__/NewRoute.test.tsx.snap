// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewRoutde renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col sm:flex-row p-2"
  >
    <div
      class="basis-4/6"
    />
    <div
      class="basis-2/6 flex flex-col p-2"
    >
      <div
        class="flex justify-end"
      >
        <button
          class="bg-brand text-white
 py-2 px-4 rounded-sm hover:brightness-110"
        >
          삭제
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NewRoute renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col sm:flex-row p-2"
  >
    <div
      class="basis-4/6"
    />
    <div
      class="basis-2/6 flex flex-col p-2"
    >
      <div
        class="flex justify-end"
      >
        <button
          class="bg-brand text-white
 py-2 px-4 rounded-sm hover:brightness-110"
        >
          삭제
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
